src/coreclr/vm/runtimehandles.cpp

```diff
@@ -1143,6 +1143,45 @@ FCIMPL2(FC_BOOL_RET, RuntimeTypeHandle::CompareCanonicalHandles, ReflectClassBas
 }
 FCIMPLEND
 
+FCIMPL1(Object*, RuntimeTypeHandle::InternalAllocNoChecks_FastPath, MethodTable* pMT)
+{
+    FCALL_CONTRACT;
+
+    _ASSERTE(pMT != nullptr);
+
+    if (!GCHeapUtilities::UseThreadAllocationContexts())
+    {
+        return NULL;
+    }
+
+#ifdef FEATURE_64BIT_ALIGNMENT
```

Comment:
```suggestion
    if (pMT->HasFinalizer())
    {
        return NULL;
    }

#ifdef FEATURE_64BIT_ALIGNMENT
```
We should take the slow path for finalizer too. (This is for completeness - I am not sure whether the existing uses can hit this.)